tensorflow_probability/python/positive_semidefinite_kernels/dot_product.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2018 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""Linear kernel."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow_probability.python.internal import dtype_util
+from tensorflow_probability.python.positive_semidefinite_kernels import positive_semidefinite_kernel as psd_kernel
+from tensorflow_probability.python.positive_semidefinite_kernels.internal import util
+
+__all__ = [
+  'Linear',
+]
+
+
+def _validate_arg_if_not_none(arg, assertion, validate_args):
+  if arg is None:
+    return arg
+  with tf.control_dependencies([assertion(arg)] if validate_args else []):
+    result = tf.identity(arg)
+  return result
+
+
+class Linear(psd_kernel.PositiveSemidefiniteKernel):
+  """Linear Kernel.
+
+    Is based on the dot product covariance function and can
+    be obtained from linear regression. This kernel, when
+    parameterizing a Gaussian Process, results in random functions that are
+    following a linear function.
+    In practice this kernel is mostly used in combination with the
+    exponentiated quadratic kernel to make it non-stationary and local.
+
+    ```none
+    k(x, y) = amplitude**2 * ( bias_variance**2 + x \cdot x')
+    ```
+  """
+  def __init__(self,
+               amplitude=None,
+               bias_variance=None,
+               feature_ndims=1,
+               validate_args=False,
+               name='Linear'):
+    """Construct a Linear kernel instance.
+
+    Args:
+      amplitude: Positive floating point `Tensor` that controls the maximum
```

Comment:
For the arguments, can you add a line below each one with default values?

Default Value: None (for instance).